docs/modules/api_gateway.md

```diff
@@ -0,0 +1,102 @@
+    "categories" = {
+      method = "GET"
+      type = "HTTP_PROXY"
+      proxy_url = "https://api.notifications.rackspace.com/categories"
+    }
+
+    # Send GET requests to /health to the "rest_api_handler" lambda (bypassing the authorizer)
+    "health" = {
+      method     = "GET"
+      lambda_key = "rest_api_handler"
+    }
+
+    # Send all unmatched requests to the "rest_api_handler" lambda (after passing through the authorizer)
+    "{proxy+}" = {
+      method         = "ANY"
+      authorizer_key = "authorizer_lambda"
+      lambda_key     = "rest_api_handler"
+    }
+  }
+
+}
+
+```
+
+## Argument Reference
+
+---
+
+The following arguments are supported:
+
+* `name` - The API Gateway name. Also added as a partial name to a log role and execution permission.
+* `description` - The API Gateway description.
+* `stage_name` - API stage name, applied as suffix to log group name and passed as the `STAGE_NAME` variable. (Use this for frameworks that need to know the _base_path_.)
+* `endpoint_type` - (optional, default "EDGE") - Endpoint configuration type. ["REGIONAL"](https://docs.aws.amazon.com/apigateway/latest/developerguide/create-regional-api.html) or ["EDGE"](https://docs.aws.amazon.com/apigateway/latest/developerguide/create-api-resources-methods.html)
+* `root_route` - A single `route` mapping specifying the behavior of the root ("/") endpoint.
+* `routes` - A mapping of path prefixes to `route` mappings to define the behavior at that path prefix.
+* `tags` - (optional) A mapping of tags to be applied to all resources.
+* `lambdas` - A mapping of string keys to an attribute mapping. The keys are arbitrary for reference in `root_route` and `route` entries.
+
+The `lambda` attribute map contains:
+
+* `function_name` - the function name of the lambda to be called
+* `authorizer_result_ttl_in_seconds` - (optional, default to "900") ttl in seconds for an authorizer result.
+* `identity_source` - (optional, default "method.request.header.X-Auth-Token") identity source for an authorizer lambda.
+
+A `route` mapping can contain:
+
+* `method` - API-Gateway [http method](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_method#http_method) default: ANY
+* `type` - API Gateway [integration type](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_integration#type). Only "AWS_PROXY" and "HTTP_PROXY" supported currently. Default: "AWS_PROXY"
+* `proxy_url` - if type is "HTTP_PROXY", proxy requests to this endpoint (otherwise this argument is ignored)
+* `lambda_key` - if type is "AWS_PROXY", send requests to the lambda assigned to this arbitrary key in the `lambdas` argument (otherwise argument is ignored).
+* `authorizer_key` - (optional) if type is "AWS_PROXY", add this authorizer to the route (otherwise argument is ignored).
+
+
+## Attributes Reference
+
+---
+
+In addition to all arguments above, the following attributes are exported:
```

Comment:
It might be nice to have the visualization of it but I'm not sure how useful it would be. It would just be "//{module.api_gateway.hostname}/{var.stage_name}". I can add it in.

Reply:
Added as "stage_url"